.github/generate-doc.py

```diff
@@ -0,0 +1,168 @@
+import glob
+import os
+import shutil
+import subprocess
+
+
+class Colors:
+    RED = '\033[91m'
+    GREEN = '\033[92m'
+    YELLOW = '\033[93m'
+    BLUE = '\033[94m'
+    RESET = '\033[0m'
+
+
+def auto_doc_installed():
+    auto_doc_cmd = os.environ.get("DOC_CMD")
+    try:
+        subprocess.run(
+            auto_doc_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, check=True)
+        return True
+    except subprocess.CalledProcessError as e:
+        return f"Error: {e.stderr}"
+
+
+def print_colored(text, color):
+    print(f"{color}{text}{Colors.RESET}")
+
+
+class DocGenerationError(Exception):
+    def __init__(self):
+        self.message = "Error: The documentation is not up to date. Re running pre-commit may help."
+        super().__init__(f"{Colors.RED}{self.message}{Colors.RESET}")
+
+
+def safe_remove_directory(directory_path):
+    if os.path.isdir(directory_path):
+        try:
+            shutil.rmtree(directory_path)
+        except Exception as e:
+            print(f"Error while removing directory '{directory_path}': {e}")
+    else:
+        print(f"Directory '{directory_path}' does not exist.")
+
+
+def copy_file(source_path, destination_path):
+    try:
+        shutil.copy(source_path, destination_path)
+    except FileNotFoundError:
+        print_colored(f"Source file not found.", Colors.RED)
+    except PermissionError:
+        print_colored(
+            "Permission error. Make sure you have the necessary permissions.", Colors.RED)
+    except Exception as e:
+        print_colored(f"An error occurred: {e}", Colors.RED)
+
+
+def remove_formatting(content):
+    # Remove whitespaces and newlines and hyphens
+    return content.replace(" ", "").replace("\n", "").replace("-", "")
+
+
+def files_equal(file1_path, file2_path):
+    with open(file1_path, 'r') as file1, open(file2_path, 'r') as file2:
+        content1 = remove_formatting(file1.read())
+        content2 = remove_formatting(file2.read())
+
+        return content1 == content2
+
+
+def run():
+    auto_doc_cmd = os.environ.get("DOC_CMD")
```

Comment:
I like the introduction of an env variable here to also enable CI runs, but can we set the default to `auto-doc` if it is empty so it works locally without the need to manually set the variable?